### ISSUE TITLE
Connection not dropped and sync stalled when sharing folder first, and only then adding password

### ISSUE THREAD

**Issue by tomasz1986** [syncthing/syncthing#7921]
1. On Device 1, share a folder with Device 2.
  ![image](https://user-images.githubusercontent.com/5626656/131617478-90083c56-915f-4425-8564-ab3ca6a5f2c6.png)
  ![image](https://user-images.githubusercontent.com/5626656/131617485-205a6ded-b3b1-418a-b356-243eb05eaa9b.png)
2. Only now, set a password for the folder on Device 1.
  ![image](https://user-images.githubusercontent.com/5626656/131617579-123fffd6-1c1f-4714-9e18-280be2ed2b67.png)
3. On Device 2, try to add the folder. The type is correctly detected and automatically set to Receive Encrypted.
  ![image](https://user-images.githubusercontent.com/5626656/131617763-3801913c-3cf3-4b65-aace-99603e51ef05.png)
4. On Device 2, the folder has been added and is marked as "Up to Date". On Device 1, the files are stuck in sync though. The connection itself has not been dropped after adding the folder, as should normally happen when adding a new Receive Encrypted folder.
  ![image](https://user-images.githubusercontent.com/5626656/131617979-44d0945c-f1b5-42c0-b122-95e5f1adad52.png)
  ![image](https://user-images.githubusercontent.com/5626656/131617984-7401dd58-3936-406c-b075-9e0f8b722be7.png)

The workaround is to manually disconnect and reconnect the two devices, which will then force the folder to sync.

**Comment by calmh** [comment]
I started trying to reproduce this, and then thought "actually, no". What you've done here is in step one share a folder unencrypted with device two. By sheer "luck" (since it's you on the other side as well) the folder wasn't accepted, otherwise it would have already started syncing a bunch of clear text data. Then you change the folder to be shared encrypted. This is super-duper not supported, crazy dangerous (assuming the data was supposed to be confidential to begin with) and should probably be blocked in the GUI to begin with...

So there's a bug here, but I don't think it is that what you did should work. Rather it's that what you did shouldn't be possible, or should be associated with big warning flags that you need to physically visit the device that was previously shared plaintext and nuke it from orbit to ensure there's no plaintext data...

**Comment by tomasz1986** [comment]
Yeah, the problem here is that the user may share the folder without setting up a password first (by mistake) and only then realise that they want to share it protected with a password. What should be the proper solution here? This is assuming that the other side still hasn't accepted the folder.

**Comment by tomasz1986** [comment]
Is this issue possibly related to https://github.com/syncthing/syncthing/issues/8041? Or, are they maybe about one and the same problem? 😇

**Comment by calmh** [comment]
Possibly, but they're conceptually different and as noted above I think should probably be solved differently...
